subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/BuildScriptBuilder.java

```diff
@@ -985,21 +1021,36 @@ public void writeBodyTo(PrettyPrinter printer) {
 
     private static class MavenRepoExpression extends AbstractStatement {
         private final String url;
+        private final BuildScriptBuilder builder;
 
-        MavenRepoExpression(@Nullable String comment, String url) {
+        MavenRepoExpression(@Nullable String comment, String url, BuildScriptBuilder builder) {
             super(comment);
             this.url = url;
+            this.builder = builder;
         }
 
         @Override
         public void writeCodeTo(PrettyPrinter printer) {
             ScriptBlockImpl statements = new ScriptBlockImpl();
-            statements.propertyAssignment(null, "url", new MethodInvocationExpression(null, "uri", Collections.singletonList(new StringValue(url))), true);
+            ensureProtocolSecurity(url, statements);
             printer.printBlock("maven", statements);
         }
+
+        private void ensureProtocolSecurity(String url, ScriptBlockImpl statements) {
+            final URL urlAsURL = urlFromString(url);
+            final Protocol protocol = Protocol.fromUrl(urlAsURL);
+
+            if (protocol.isSecure()) {
+                statements.propertyAssignment(null, "url", new MethodInvocationExpression(null, "uri", Collections.singletonList(new StringValue(url))), true);
+            } else {
+                final Logger logger = LoggerFactory.getLogger(BuildScriptBuilder.class);
+                logger.warn("Repository URL: '{}' uses an insecure protocol.", url);
+                builder.insecureProtocolHandler.handle(urlAsURL, statements);
+            }
```

Comment:
I think we can replace `Protocol` with just `GUtil.isSecureUrl` and `GUtil.toSecureUrl`. e.g.,

```suggestion
            final URI urlAsURI = urlFromString(url);

            if (GUtil.isSecureUrl(urlAsURI)) {
                statements.propertyAssignment(null, "url", new MethodInvocationExpression(null, "uri", Collections.singletonList(new StringValue(url))), true);
            } else {
                final Logger logger = LoggerFactory.getLogger(BuildScriptBuilder.class);
                logger.warn("Repository URL: '{}' uses an insecure protocol.", url);
                builder.insecureProtocolHandler.handle(urlAsURL, statements);
            }
```

One reason to do this is that it keeps the checks that the insecure protocol code does the same on both sides.